gradio/wasm_utils.py

```diff
@@ -16,6 +16,8 @@ class WasmUnsupportedError(Exception):
 # the Gradio's FastAPI app instance (`app`).
 def register_app(_app):
     global app
+    if app:
+        app.blocks.close()
```

Comment:
I'm not sure what this is doing since `app = None` above. Am I missing something?

Reply:
`register_app()` is called repeatedly, every time a user runs new code with `run_file()` or `run_code()`.

So,
* `app = None` is an initialization of the `app` object.
* This `if app` can be true in the second call and later.

Reply:
Got it, thanks for clarifying